core/domain/activity_jobs_one_off.py

```diff
@@ -246,3 +246,230 @@ def reduce(key, values):
+            elif commit_cmd['cmd'] == feconf.CMD_REMOVE_MANAGER_ROLE:
+                mentioned_user_ids.add(commit_cmd['removed_user_id'])
+            snapshot_model.mentioned_user_ids = list(
+                sorted(mentioned_user_ids))
+            snapshot_model.put(update_last_updated_time=False)
+
+    @classmethod
+    def enqueue(cls, job_id, additional_job_params=None):
+        # We can raise the number of shards for this job, since it goes only
+        # over three types of entity class.
+        super(AddMentionedUserIdsMetadataJob, cls).enqueue(
+            job_id, shard_count=64)
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        """Return a list of datastore class references to map over."""
+        return [collection_models.CollectionRightsSnapshotMetadataModel,
+                exp_models.ExplorationRightsSnapshotMetadataModel,
+                topic_models.TopicRightsSnapshotMetadataModel]
+
+    @staticmethod
+    def map(snapshot_model):
+        """Implements the map function for this job."""
+        class_name = snapshot_model.__class__.__name__
+        if isinstance(
+                snapshot_model,
+                collection_models.CollectionRightsSnapshotMetadataModel):
+            AddMentionedUserIdsMetadataJob._add_col_and_exp_user_ids(
+                snapshot_model)
+        elif isinstance(
+                snapshot_model,
+                exp_models.ExplorationRightsSnapshotMetadataModel):
+            AddMentionedUserIdsMetadataJob._add_col_and_exp_user_ids(
+                snapshot_model)
+        elif isinstance(
+                snapshot_model,
+                topic_models.TopicRightsSnapshotMetadataModel):
+            AddMentionedUserIdsMetadataJob._add_topic_user_ids(snapshot_model)
+        yield ('SUCCESS-%s' % class_name, snapshot_model.id)
+
+    @staticmethod
+    def reduce(key, ids):
+        """Implements the reduce function for this job."""
+        yield (key, len(ids))
+
+
+class AuditSnapshotMetadataModelsJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that audits commit_cmds field of the snapshot metadata models. We log
+    the length of the commit_cmd, the possible 'cmd' values, and all the other
+    keys.
+    """
+
+    @classmethod
+    def enqueue(cls, job_id, additional_job_params=None):
+        # We can raise the number of shards for this job, since it goes only
+        # over three types of entity class.
+        super(AuditSnapshotMetadataModelsJob, cls).enqueue(
+            job_id, shard_count=64)
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        """Return a list of datastore class references to map over."""
+        return [collection_models.CollectionRightsSnapshotMetadataModel,
+                exp_models.ExplorationRightsSnapshotMetadataModel,
+                topic_models.TopicRightsSnapshotMetadataModel]
+
+    @staticmethod
+    def map(snapshot_model):
+        """Implements the map function for this job."""
```

Comment:
Do you need to check that the entity isn't deleted yet?

Reply:
Done.